lib/licensed/commands/status.rb

```diff
@@ -35,20 +35,52 @@ def evaluate_dependency(app, source, dependency, report)
         else
           report.errors << "cached dependency record out of date" if cached_record["version"] != dependency.version
           report.errors << "missing license text" if cached_record.licenses.empty?
-          report.errors << "license needs review: #{cached_record["license"]}" unless allowed_or_reviewed?(app, cached_record)
+          report.errors << "license needs review: #{cached_record["license"]}" if license_needs_review?(app, cached_record)
         end
 
         report.errors.empty?
       end
 
-      def allowed_or_reviewed?(app, dependency)
-        app.allowed?(dependency) || app.reviewed?(dependency)
+      # Returns true if a cached record needs further review based on the
+      # record's license(s) and the app's configuration
+      def license_needs_review?(app, cached_record)
+        # review is not needed if the record is set as reviewed
+        return false if app.reviewed?(cached_record)
+        # review is not needed if the top level license is allowed
+        return false if app.allowed?(cached_record["license"])
+
+        # the remaining checks are meant to allow records marked as "other"
+        # that have multiple licenses, all of which are allowed
+
+        # review is needed for non-"other" licenses
+        return true unless cached_record["license"] == "other"
+
+        licenses = cached_record.licenses.map { |license| license_from_text(license.text) }
```

Comment:
Ignorant question: can a license detected from a package manager file be in this list? Asking because it wouldn't be detected by `license_from_text` below.

Reply:
👍 it won't be detected because package manager text [isn't cached](https://github.com/github/licensed/blob/4abf047b99910f5a28a93f599381ab5bd7cdf8dc/lib/licensed/dependency.rb#L75) under the `licenses` field.

That in itself is somewhat confusing as it's a one-time / hidden check.  That's been around since before I started maintaining licensed, but I'll log an issue to re-examine if we should track that in the cached data and the best way to represent that.

Reply:
https://github.com/github/licensed/issues/189